Sources/ApolloCodegenLib/ApolloSchemaDownloadConfiguration.swift

```diff
@@ -0,0 +1,90 @@
+import Foundation
+
+/// A configuration object that defines behavior for schema download.
+public struct ApolloSchemaDownloadConfiguration {
+  
+  /// How to attempt to download your schema
+  public enum DownloadMethod: Equatable {
+
+    case registry(_ settings: RegistrySettings)
+    ///   - endpointURL: The endpoint to hit to download your schema.
+    case introspection(endpointURL: URL)
```

Comment:
Can we document these two options and what they mean. I'm not even sure I know how to use a registry. For new GraphQL users, they may not even know what a registry IS.

Reply:
The registry is the Apollo registry. This *might* be one case where we could put `ApolloRegistrySettings` in for clarity

Reply:
Yeah, the Registry is a service provided by Apollo as a company, I think it makes sense to refer to it as apollo registry. 

I'm thinking the `RegistrySettings` is fine to stay as it. I'd actually prefer to rename the enum case to `case apolloRegistry(_ settings: RegistrySettings)`

We should still document this stuff a bit better.

Reply:
Good point on the "Apollo" prefix. I'll add that and add some documentation for the two values.

Reply:
It should make more sense for new GraphQL users with 7b0c7779.